freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -350,10 +351,15 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
                 LOGGER.debug("Finished {}.", LogEvent.SET_WORKLOAD_CREDENTIALS);
             }
 
-            // TODO For now we only sync cloud ids during full sync. We should eventually allow more granular syncs (actor level and group level sync).
-            if (fullSync && entitlementService.cloudIdentityMappingEnabled(INTERNAL_ACTOR_CRN, stack.getAccountId())) {
+            if (entitlementService.cloudIdentityMappingEnabled(INTERNAL_ACTOR_CRN, stack.getAccountId())) {
                 LOGGER.debug("Starting {} ...", LogEvent.SYNC_CLOUD_IDENTITIES);
-                cloudIdentitySyncService.syncCloudIdentities(stack, umsUsersState, warnings::put);
+                // For cloud identites we only allow full sync or syncing a single user
+                if (fullSync) {
+                    cloudIdentitySyncService.syncCloudIdentities(stack, umsUsersState, warnings::put);
+                } else if (umsUsersState.getUserToCloudIdentityMap().size() == 1) {
```

Comment:
what if we have more than one? since user sync supports sets of actors.

Reply:
As per comment "For cloud identities we only allow full sync or syncing a single user". The public interfaces don't allow syncing sets of actors so I didn't think we want/need internal support for sets of actors?